src/main/groovy/life/qbic/registration/MeasurementID.groovy

```diff
@@ -0,0 +1,126 @@
+package life.qbic.registration
+
+import groovy.transform.EqualsAndHashCode
+import life.qbic.registration.types.QDatasetType
+
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+/**
+ * A measurement ID references an openBIS object code used for measurements
+ *
+ * @since 1.0.0
+ */
+@EqualsAndHashCode
+class MeasurementID {
+
+    /**
+     * <p>The regular expression of a valid QBiC measurement code.</p>
+     */
+    static final Pattern QBIC_MEASUREMENT_CODE_SCHEMA = ~/N*[GM]SQ[A-X0-9]{4}[0-9]{3}[A-X0-9]{2}-[0-9]{14}/
```

Comment:
NIT: IMHO this regex is not extendible for the different measurementIds we want to support. We currently have already 2 MeasurementId Prefixes (NGS/MS). Can the regex be a combination of individual regexes validating for each individual measurementId?